pkg/server/plugin/nodeattestor/awsiid/organization.go

```diff
@@ -0,0 +1,254 @@
+}
+
+type orgValidator struct {
+	orgListAccountMap             map[string]bool
+	orgListAccountMapCreationTime time.Time
+	orgConfig                     *orgValidationConfig
+	mutex                         sync.RWMutex
+	orgAccountListCacheTTL        time.Duration // set from configuration else will be set to default
+	log                           hclog.Logger
+	retries                       int // fix number of retries before ttl is expired.
+}
+
+func newOrganizationValidationBase(config *orgValidationConfig) *orgValidator {
+	client := &orgValidator{
+		orgListAccountMap: make(map[string]bool),
+		orgConfig:         config,
+		retries:           orgAccountRetries,
+	}
+
+	return client
+}
+
+func (o *orgValidator) getRetries() int {
+	o.mutex.RLock()
+	defer o.mutex.RUnlock()
+	return o.retries
+}
+
+func (o *orgValidator) decrRetries() int {
+	o.mutex.Lock()
+	defer o.mutex.Unlock()
+	if o.retries > 0 {
+		o.retries -= 1
+	}
+
+	return o.retries
+}
+
+func (o *orgValidator) configure(config *orgValidationConfig) error {
+	o.mutex.Lock()
+	defer o.mutex.Unlock()
+
+	o.orgConfig = config
+	// required for the way existing getClient function is designed
+	o.orgConfig.AccountRegion = "us-west-2"
+
+	// While doing configuration invalidate the map so we dont keep using old one.
+	o.orgListAccountMap = make(map[string]bool)
+	o.retries = orgAccountRetries
+
+	t, err := time.ParseDuration(config.AccountListTTL)
+	if err != nil {
+		return status.Errorf(codes.Internal, "issue while parsing ttl for organization, while configuring orgnization validation: %v", err)
+	}
+
+	o.orgAccountListCacheTTL = t
+
+	return nil
+}
+
+func (o *orgValidator) setLogger(log hclog.Logger) {
+	o.log = log
+}
+
+// This verification method checks if the Account ID attached on the node is part of the organisation.
+// If it part of the organisation then validation should be succesfull if not attestation should fail, on enabling this verification method.
+// This could be alternative for not explictly maintaing allowed list of account ids.
+// Method pulls the list of accounts from organization and caches it for certain time, cache time can be configured.
+func (o *orgValidator) IsMemberAccount(ctx context.Context, orgClient organizations.ListAccountsAPIClient, accoundIDofNode string) (bool, error) {
+
+	reValidatedcache, err := o.validateCache(ctx, orgClient)
+	if err != nil {
+		return false, err
+	}
+
+	accountIsmemberofOrg, err := o.lookupCache(ctx, orgClient, accoundIDofNode, reValidatedcache)
+	if err != nil {
+		return false, err
+	}
+
+	return accountIsmemberofOrg, nil
+}
+
+// Validate cache and refresh if its stale
+func (o *orgValidator) validateCache(ctx context.Context, orgClient organizations.ListAccountsAPIClient) (bool, error) {
+
+	isStale, err := o.checkIfOrgAccountListIsStale(ctx)
+	if err != nil {
+		return isStale, err
+	}
+
+	// refresh the account map
+	if isStale {
+		_, err = o.reloadAccountList(ctx, orgClient, false)
```

Comment:
when it is stale, but you reload, depending if it return error or no, isStale value keep the same, is that ok? 
we can reload and return it as stale but it was reloaded,
or if it fails it will be true, but reload failed